tianshou/policy/base.py

```diff
@@ -300,8 +301,8 @@ def compute_action(
         :param state: the hidden state of RNN policy, used for recurrent policy.
         :return: action as int (for discrete env's) or array (for continuous ones).
         """
-        # need to add empty batch dimension
-        obs = obs[None, :]
+        obs = np.array(obs)  # convert array-like to array (e.g. LazyFrames)
+        obs = obs[None, :]  # add batch dimension
```

Comment:
I think the original intention here I made is to allow any Batch-like obs in this code path, so `batch[None, :]` will add batch-size dimension to all nested tree elements

Reply:
@MischaPanch added the `compute_action` method to have a convenient interface for the inference case (i.e. when actually applaying the agent in practice), so we need a function that can handle the observations the way they are produced by the environment. The operation in line 305, however, will fail for things like `LazyFrames` if we don't ensure that array-like representations are converted to actual arrays first.

Reply:
Indeed, it's currently unused. Examples will follow soon in udated tutorials.

Thanks @opcode81 ! Merging this